### PR TITLE
Fix org confirmation link and other fixes

### PR DESCRIPTION
**Pull request by chenriksson**
Assorted fixes for Organizations:
- Fixed confirmationUrl in email sent on changing Organization email address
- Allow current user to modify self (if admin) in Manage Organization
- Allow collaborators to view (not modify) data in Manage Organization

Screenshot - collaborator's view:
- disabled sections, but showing the data
- Members section displays members but is readonly

image

### CODE REVIEW

File: src/NuGetGallery/App_Code/ViewHelpers.cshtml [hunk 1]
```diff
@@ -515,29 +517,45 @@ var hlp = new AccordionHelper(name, formModelStatePrefix, expanded, page);
...
             bool expanded = false,
             string expandedIcon = "ChevronDown",
-            string collapsedIcon = "ChevronRight")
+            string collapsedIcon = "ChevronRight",
+            string disabledIcon = "Lock",
+            bool disabled = false)
 {
-    AddSection(viewPage.ViewBag, id);
+    if (!disabled)
+    {
```
Comment (cristinamanum) [inline src/NuGetGallery/App_Code/ViewHelpers.cshtml]: It would be great to add a snapshot of the ui.

Comment (chenriksson) [reply src/NuGetGallery/App_Code/ViewHelpers.cshtml]: @cristinamanum thanks - I thought I did, but it was to the wrong PR Fixed!

File: src/NuGetGallery/Controllers/AccountsController.cs [hunk 2]
```diff
@@ -283,7 +287,7 @@ protected virtual TUser GetAccount(string accountName)
         protected virtual ActionResult AccountView(TUser account, TAccountViewModel model = null)
         {
             if (account == null
-                || ActionsRequiringPermissions.ManageAccount.CheckPermissions(GetCurrentUser(), account)
+                || ActionsRequiringPermissions.ViewAccount.CheckPermissions(GetCurrentUser(), account)
```
Comment (cristinamanum) [inline src/NuGetGallery/Controllers/AccountsController.cs]: In which cases an account twill have only view but not update permissions?

Comment (chenriksson) [reply src/NuGetGallery/Controllers/AccountsController.cs]: This action is for both Account.cshtml (User) and ManageOrganization.cshtml (Organization). For organizations, an admin can manage (read-write) the account, and a collaborator can view (read-only)

**Comment by skofman1** [comment]
@chenriksson , is this still relevant (it's from before the bug bash)?

**Comment by chenriksson** [comment]
@skofman1 Yes - still relevant and very important.

The bug bash was from this feature branch, which included latest in dev at the time.

File: src/NuGetGallery/Controllers/AccountsController.cs [hunk 3]
```diff
@@ -85,7 +85,9 @@ public virtual ActionResult ConfirmationRequiredPost(string accountName = null)
                 return new HttpStatusCodeResult(HttpStatusCode.Forbidden, Strings.Unauthorized);
             }
 
-            var confirmationUrl = Url.ConfirmEmail(account.Username, account.EmailConfirmationToken, relativeUrl: false);
+            var confirmationUrl =  (account is Organization)
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/AccountsController.cs]: This is an abstract class. It shouldn't have info about the account type (org or user). Add an abstract method to get Confirm mail from child class.

File: src/NuGetGallery/Controllers/AccountsController.cs [hunk 4]
```diff
@@ -236,7 +238,9 @@ public virtual ActionResult ConfirmationRequiredPost(string accountName = null)
 
             if (account.Confirmed)
             {
-                var confirmationUrl = Url.ConfirmEmail(account.Username, account.EmailConfirmationToken, relativeUrl: false);
+                var confirmationUrl = (account is Organization)
+                    ? Url.ConfirmOrganizationEmail(account.Username, account.EmailConfirmationToken, relativeUrl: false)
```
Comment (skofman1) [inline src/NuGetGallery/Controllers/AccountsController.cs]: same

File: src/NuGetGallery/Services/ActionsRequiringPermissions.cs [hunk 5]
```diff
@@ -94,5 +94,13 @@ public static class ActionsRequiringPermissions
         public static ActionRequiringAccountPermissions ManageAccount =
             new ActionRequiringAccountPermissions(
                 accountPermissionsRequirement: RequireOwnerOrOrganizationAdmin);
+
+        /// <summary>
+        /// The action of managing a user or organization account. This includes confirming an account,
+        /// changing the email address, changing email subscriptions, modifying sign-in credentials, etc.
```
Comment (skofman1) [inline src/NuGetGallery/Services/ActionsRequiringPermissions.cs]: the comment doesn't match what the code does

**Pull request merged**